nntrainer/graph/graph_node.h

```diff
@@ -117,7 +117,7 @@ class GraphNodeIterator
    * @return value_type
    * @note this is different from standard iterator
    */
-  value_type operator*() const {
+  const value_type operator*() const {
```

Comment:
this shouldn't be needed. If you want to use const iterator, use `graph_const_iterator` which automatically inserts const here.

Reply:
Deleted